test/thirdparty.test.ts

```diff
@@ -0,0 +1,227 @@
+        {
+          name: "Bobby Tables",
+          email: "bob-tables@gmail.com",
+        },
+        {
+          name: "Jane Tables",
+          email: "janet@gmail.com",
+        },
+      ]);
+
+      await result.meta.txn.wait();
+
+      const { results } = await users.All({
+        where: { name: "Bobby Tables" },
+        limit: 1,
+        offset: 0,
+        orderBy: ["id"],
+      });
+
+      deepStrictEqual(results, [
+        {
+          name: "Bobby Tables",
+          id: 1,
+          email: "bobby-tab@gmail.com",
+        },
+      ]);
+    });
+
+    test("basic query building works well to then query the data", async function () {
+      const { query, bindings } = GenerateQuery<User>(
+        QueryType.SELECT,
+        users.tableName, // Could also come from the above meta.txn objects
+        {
+          where: {
+            name: "Bobby Tables",
+          }, // this uses the type from above to enforce it to properties which exist on the table
+          limit: 1, // we only want the first user
+          offset: 1, // skip the first user named 'Bobby Tables' when performing this query
+
+          // Using orderBy is a special case, so there's a few possible syntaxes for it
+          orderBy: { column: "id", descending: true }, // ORDER BY id DESC NULLS LAST
+        }
+      );
+
+      // Using the database directly
+      const stmt = db.prepare(query).bind(bindings);
+      const { results } = await stmt.all();
+      deepStrictEqual(results, [
+        {
+          name: "Bobby Tables",
+          id: 1,
+          email: "bobby-tab@gmail.com",
+        },
+      ]);
+    });
+
+    test("where upserts are easier when using an orm", async function () {
+      const user: User = {
+        id: 1,
+        name: "John Doe",
+        email: "john-doe@gmail.com",
+      };
+      const { query, bindings } = GenerateQuery<User>(
+        QueryType.UPSERT,
+        users.tableName, // Could also come from the above meta.txn objects
+        {
+          data: user,
+          upsertOnlyUpdateData: {
+            name: user.name,
+            email: user.email,
+          },
+          where: {
+            id: user.id,
+          },
+        },
+        "id"
+      );
+      // {
+      //    query: "INSERT INTO users (id, name, email) VALUES (?, ?, ?) ON CONFLICT(id) DO UPDATE SET name = ?, email = ? WHERE id = ?",
+      //    bindings: [1, "John Doe", "john-doe@gmail.com", "John Doe", "john-doe@gmail.com", 1]
+      // }
+
+      // Using the database directly
+      const stmt = db.prepare(query).bind(bindings);
+      const { meta } = await stmt.run();
+      const receipt = await (meta.txn as any).wait();
+      strictEqual(receipt?.error, undefined);
+
+      const results = await db
+        .prepare(`SELECT * FROM ${users.tableName} WHERE id=?`)
+        .bind(user.id)
+        .first<User>();
+      deepStrictEqual(results, user);
+    });
+  });
+
+  describe("@databases/sql", function () {
```

Comment:
This is a lib that I've seen others using. It is quite nice, and has lots of interesting functionality. In particular, using tagged template literals for working with SQL is really nice. I'd go so far as suggesting we recommend this in our docs. I'll mention why in a sec.